### ISSUE TITLE
[a11y]: Datatable Cells not readable in Screenreader in Carbon

### ISSUE THREAD

**Issue by RheaM29** [carbon-design-system/carbon#20061]
### Package

carbon-components

### Package version

v11

### Automated testing tool and ruleset

Voice Over for Mac

### Assistive technology

Voice Over

### Description

The IBM Accessibility Website does not provide any information around Datatable cells being read by Screenreader. There is no guidance around Screenreader being able to read all the cells in the datatable or not. 
Ideally for a visually impaired user, the screen reader should read all text in every cell, but Carbon does not seem to have rules around it on the website- [Datatables section](https://www.ibm.com/able/toolkit/design/visual/#data-tables).
A visually disabled user relies only on the screen reader to read every word on the page. This violates [WCAG guidelines](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html). 

In the screenshot from IBM Concert example below, the Screenreader reads the titles then reads the contents in the 'Name' and 'Published' columns only- and calls it a 'Button'.

<img width="1383" height="614" alt="Image" src="https://github.com/user-attachments/assets/601bd190-666c-4a65-b22d-27cdf2b4bc96" />

### Reproduction/example

NA

### Steps to reproduce

- Open the "Datatable" component
- Notice that the screen reader reads only columns with Buttons, Links and Icons
- It skips reading the other columns of the datatable

This behaviour can be seen in all datatables within IBM Concert, IBM Instana and some other IBM products as well. 

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
@RheaM29 Hi, could you reproduce the issue in a [stackblitz](https://stackblitz.com/github/carbon-design-system/sandboxes/tree/main/react/latest/react-19-vite?file=src%2Findex.scss,src%2FApp.jsx&preset=node=), and post the link back here? It's hard to know exactly what's going wrong without seeing the issue firsthand. Our most recent manual testing with a real screenreader user did not show any defects with the [Carbon DataTable examples](https://react.carbondesignsystem.com/?path=/story/components-datatable-basic--default) present in the storybook.

**Comment by RheaM29** [comment]
https://github.com/user-attachments/assets/544304f7-b7e5-4a6a-a5b5-2b77a1b8fae0

@tay1orjones I am unable to reproduce the issue in stackblitz. Hope this screen recording helps!

**Comment by tay1orjones** [comment]
It looks like you're navigating using <kbd>tab</kbd> which is only going to focus interactive elements. To read non-interactive table content you have to specifically invoke voiceover to do so. See here:

* [Use VoiceOver to browse webpages on Mac](https://support.apple.com/guide/voiceover/browse-webpages-vo27974/10/mac/15.0)
* [Use the VoiceOver rotor on Mac](https://support.apple.com/guide/voiceover/with-the-voiceover-rotor-mchlp2719/10/mac/15.0)
* [Use VoiceOver to navigate webpage tables on Mac](https://support.apple.com/guide/voiceover/navigate-webpage-tables-vo27958/mac)

In the following video I use <kbd>caps lock</kbd> + <kbd>U</kbd> to open the rotor, select the table, and then use the instructed keypress combinations to navigate. You can un-mute the video to hear how it's read.

https://github.com/user-attachments/assets/a55c43ed-d613-4efe-8f73-61b6fe31ef6f

**Comment by RheaM29** [comment]
This helps @tay1orjones! Thank you!
